File: Projects/Framework-Samples/PLINQO/VisualBasic/Tracker/Source/Tracker.Tests/QueryTests/ByDateTimeTests.vb
Imports System.Collections.Generic
Imports System.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.QueryTests
    <TestFixture()> _
    Public Class ByDateTimeTests
        Inherits TestBase
        <Test()> _
        Public Sub ByTest()
            Using db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }
                Dim a As List(Of Task) = db.Task.ByCreatedDate(CreateDates(0)).ToList()
                Dim b As List(Of Task) = db.Task.ByCreatedDate(CreateDates(1)).ToList()

                'Dim c As List(Of Task) = db.Task.ByCreatedDate(CreateDates(0), CType(Nothing, System.Nullable(Of DateTime))).ToList()
                'Assert.AreEqual(a.Count, c.Count)
            End Using
        End Sub

        <Test()> _
        Public Sub ByNullableTest()
            Using db As New TrackerDataContext() With { _
             .Log = Console.Out _
            }
                Dim a As List(Of Task) = db.Task.ByDueDate(CType(Nothing, System.Nullable(Of DateTime))).ToList()
                Dim b As List(Of Task) = db.Task.ByDueDate(DueDates(0)).ToList()
                Dim c As List(Of Task) = db.Task.ByDueDate(DueDates(1)).ToList()

                Dim d As List(Of Task) = db.Task.ByDueDate(DueDates(0), CType(Nothing, System.Nullable(Of DateTime))).ToList()
                Assert.AreEqual(a.Count + b.Count, d.Count)

                Dim e As List(Of Task) = db.Task.ByDueDate(DueDates(0), CType(Nothing, System.Nullable(Of DateTime)), DueDates(1)).ToList()
                Assert.AreEqual(a.Count + b.Count + c.Count, e.Count)
            End Using
        End Sub
    End Class
End Namespace

File: Projects/Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Tests/QueryTests/ByDateTimeTests.cs
using System;
using System.Collections.Generic;
using System.Data.Linq;
using System.Linq;
using NUnit.Framework;
using Tracker.Core.Data;

namespace Tracker.Tests.QueryTests
{
    [TestFixture]
    public class ByDateTimeTests : TaskTests
    {
        [Test]
        public void ByTest()
        {

                using (var db = new TrackerDataContext())
                {
                    var a = db.Task.ByCreatedDate(createDates[0]).ToList();
                    var b = db.Task.ByCreatedDate(createDates[1]).ToList();

                    var c = db.Task.ByCreatedDate(createDates[0], null).ToList();
                    Assert.AreEqual(a.Count, c.Count);

                    var d = db.Task.ByCreatedDate(createDates[0], createDates[1]).ToList();
                    Assert.AreEqual(a.Count + b.Count, d.Count);
                }

        }

        [Test]
        public void ByNullableTest()
        {

                using (var db = new TrackerDataContext())
                {
                    var a = db.Task.ByDueDate((DateTime?)null).ToList();
                    var b = db.Task.ByDueDate(dueDates[0]).ToList();
                    var c = db.Task.ByDueDate(dueDates[1]).ToList();

                    var d = db.Task.ByDueDate(dueDates[0], null).ToList();
                    Assert.AreEqual(a.Count + b.Count, d.Count);

                    var e = db.Task.ByDueDate(dueDates[0], null, dueDates[1]).ToList();
                    Assert.AreEqual(a.Count + b.Count + c.Count, e.Count);
                }

        }
    }
}
